### PR TITLE
Docker suggestions

### PR DESCRIPTION
**Pull request by Woodham**
* Move run image to nginx base
* use envsubst to replace environment variables in config file
* create config file template
* move docker specific files to directory

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,8 +14,10 @@ COPY . .
...
-ENTRYPOINT [ "/entrypoint.sh" ]
-COPY --chown=1000 entrypoint.sh ./
-COPY --from=builder --chown=1000 /app/Lightspeed-react/build /usr/share/nginx/html
+FROM nginx:${NGINX_VERSION}-alpine
+ENV WEBSOCKET_HOST=localhost
+ENV WEBSOCKET_PORT=8080
+
+COPY docker/entrypoint.sh /docker-entrypoint.d/entrypoint.sh
+COPY docker/config.json.template /config.json.template
+COPY --from=builder /app/Lightspeed-react/build /usr/share/nginx/html
```
Comment (qdm12) [inline Dockerfile:23]: Please keep the --chown flags for the 3 COPY so the container can be run without root (for security reasons). You could add a `USER 1000` instruction at the end of the Dockerfile (was keeping it for another PR 😄)

File: docker/config.json.template [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "wsUrl": "ws://${WEBSOCKET_HOST}:${WEBSOCKET_PORT}/websocket"
```
Comment (qdm12) [inline docker/config.json.template:2]: Just thinking maybe a single `WEBSOCKET_ADDRESS` containing host:port would be nice. But that's just me nitting, the way it is is nice too.

File: docker/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+publicDir=/usr/share/nginx/html
+#host=${WEBSOCKET_HOST:-localhost}
+
+# configure ip, hardcoded to webrtc container address (8080) for now
+#sed -i "s|stream.gud.software|$host|g" "$publicDir/config.json"
```
Comment (qdm12) [inline docker/entrypoint.sh:7]: Just remove these commented lined 😉

**Review by qdm12 — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (Woodham) [reply Dockerfile:23]: Sure - makes sense to me. Basing the image on nginx does mean some more tweaking is needed to actually get it to run as non root however.

File: docker/config.json.template (hunk 2 above)
Comment (Woodham) [reply docker/config.json.template:2]: Yeah I'm not precious about the specific config options at all, I just wanted to demonstrate how multiple environment vars can be used in the template easily.

File: Dockerfile (hunk 1 above)
Comment (qdm12) [reply Dockerfile]: Yeah to be honest, my short term plan is to add a Go entrypoint to replace values in config.json and serve the files over HTTP, similar to [reactserv](https://github.com/qdm12/reactserv). That way we can run it on a Scratch Docker image without root and with only a static binary entrypoint and the static UI files (and soon with Go 1.16 we can even bundle the UI files in the binary 😄). But I'm keeping it for another PR as it's a lot of changes, I don't want reviewer to throw themselves out of the window haha.

File: docker/config.json.template (hunk 2 above)
Comment (qdm12) [reply docker/config.json.template:2]: I get your point, although (in my open source experience) you may end up with [A LOT OF ENV VARIABLES](https://github.com/qdm12/gluetun/blob/master/Dockerfile#L55-L133) so 1 is better than 2 haha!

File: docker/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+publicDir=/usr/share/nginx/html
+
+envsubst < /config.json.template > "$publicDir/config.json"
```
Comment (qdm12) [inline docker/entrypoint.sh:5]: TIL! Cool tool.

**Review by qdm12 — APPROVED** [review]

**Pull request merged**